[view-transitions] https://simple-set-demos.glitch.me/gross-cube-transition doesn't render 3d effects.

mattwoodrow opened the pull request:
#### 84c369e893d8167a747e771777a93565c0d7e9df
<pre>
[view-transitions] <a href="https://simple-set-demos.glitch.me/gross-cube-transition">https://simple-set-demos.glitch.me/gross-cube-transition</a> doesn&apos;t render 3d effects.
<a href="https://bugs.webkit.org/show_bug.cgi?id=272713">https://bugs.webkit.org/show_bug.cgi?id=272713</a>
&lt;<a href="https://rdar.apple.com/126513916">rdar://126513916</a>&gt;

Reviewed by Tim Nguyen.

ancestorLayerIsDOMParent (used for determing if &apos;transform-style: preserve-3d&apos;
from the parent should be applied) was only taking into account &apos;Element&apos; ancestors.
Add support for also checking for pseudo-element parents in the view-transition
tree.

* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-element-preserve-3d-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-element-preserve-3d-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-element-preserve-3d.html: Added.
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::ancestorLayerIsDOMParent const):
(WebCore::RenderLayer::participatesInPreserve3D const):
(WebCore::parentLayerIsDOMParent): Deleted.
* Source/WebCore/rendering/RenderLayer.h:
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateTransformFlatteningLayer):
(WebCore::ancestorLayerIsDOMParent): Deleted.
* Source/WebCore/rendering/style/RenderStyleConstants.h:
(WebCore::parentPseudoElement):
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::makeResolutionContextForPseudoElement):

Canonical link: <a href="https://commits.webkit.org/277639@main">https://commits.webkit.org/277639@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6ce8098638416987776c8c082f2aedbc43e25bc5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48023 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27228 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50941 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50710 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44081 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50330 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33115 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24720 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/39270 "Build was cancelled. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; Passed layout tests; 27 flakes 77 failures; Uploaded test results") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48606 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24901 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/41545 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20365 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22382 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/42721 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6074 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44368 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43156 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52603 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23071 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19426 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/46610 "Build was cancelled. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Passed layout tests; layout-tests (exception)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24335 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/41695 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45432 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10630 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25136 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24057 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/rendering/RenderLayer.cpp:
```diff
@@ -4084,18 +4084,28 @@ Ref<HitTestingTransformState> RenderLayer::createLocalTransformState(RenderLayer
+    case PseudoId::ViewTransitionNew: parentPseudoId = PseudoId::ViewTransitionImagePair; break;
+    case PseudoId::ViewTransitionOld: parentPseudoId = PseudoId::ViewTransitionImagePair; break;
+    default: return false;
+    }
```
Can we share this code somehow with https://searchfox.org/wubkat/rev/8d7eceb1f07dd13e4e2195f14d02fc27489a87a9/Source/WebCore/style/StyleTreeResolver.cpp#571-583

nt1m reviewed (approved):
LGTM, can we add a test?

mattwoodrow committed:
[view-transitions] https://simple-set-demos.glitch.me/gross-cube-transition doesn't render 3d effects.
https://bugs.webkit.org/show_bug.cgi?id=272713
<rdar://126513916>

Reviewed by Tim Nguyen.

ancestorLayerIsDOMParent (used for determing if 'transform-style: preserve-3d'
from the parent should be applied) was only taking into account 'Element' ancestors.
Add support for also checking for pseudo-element parents in the view-transition
tree.

* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-element-preserve-3d-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-element-preserve-3d-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-element-preserve-3d.html: Added.
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::ancestorLayerIsDOMParent const):
(WebCore::RenderLayer::participatesInPreserve3D const):
(WebCore::parentLayerIsDOMParent): Deleted.
* Source/WebCore/rendering/RenderLayer.h:
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateTransformFlatteningLayer):
(WebCore::ancestorLayerIsDOMParent): Deleted.
* Source/WebCore/rendering/style/RenderStyleConstants.h:
(WebCore::parentPseudoElement):
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::makeResolutionContextForPseudoElement):

Canonical link: https://commits.webkit.org/277639@main
